polynote-server/src/main/scala/polynote/server/SocketSession.scala

```diff
@@ -42,9 +42,9 @@ object SocketSession {
         notebooks => Some(ListNotebooks(notebooks.map(ShortString.apply)))
       }
 
-    case CreateNotebook(path, maybeContent) =>
+    case CreateNotebook(path, maybeContent, maybeTemplate) =>
       NotebookManager.assertValidPath(path) *>
-        checkPermission(Permission.CreateNotebook(path)) *> NotebookManager.create(path, maybeContent).as(None)
+        checkPermission(Permission.CreateNotebook(path)) *> NotebookManager.create(path, maybeContent, maybeTemplate).as(None)
```

Comment:
Instead of passing through the template to the NotebookManager, should we undo those changes and instead load the template from the filesystem right here, and pass it through as the `maybeContent`?

IIRC, one of the goals was that this template mechanism wouldn't necessarily have to go through the repository – that way, you could have your templates in some location where they wouldn't show up as editable notebooks in Polynote (in case that was what you wanted) _or_ you could point to some location that also shows up in Polynote.

@jonathanindig WDYT?

Reply:
Ooh, I like this idea a lot.